### PR TITLE
Add version to Job Applications

### LINKED ISSUE #2841: Add a version tag to Applications
Add a field to all applications that defines its version, probably a reference to a application_version status table. All existing applications must be set to 1, ones created with the timeline would be set to 2. This will allow us to keep using the old ui for old applications.

### PR DESCRIPTION
**Pull request by gggrant**
Resolves #2841.

### Notes:
- Do we want to tag everything v1 now, or later?

### CODE REVIEW

**Comment by Khristophor** [comment]
I think we'll want to tag everything v1 right before we roll out the new Application flow, but I'm not sure how we'll remember to do that 😄

**Comment by yonikid15** [comment]
> I think we'll want to tag everything v1 right before we roll out the new Application flow, but I'm not sure how we'll remember to do that 😄

Couldn't we just tag the applications v1 now? It wouldn't have any effect. Then when the new application flow comes out, make sure to set the version to v2.

**Comment by gggrant** [comment]
> > I think we'll want to tag everything v1 right before we roll out the new Application flow, but I'm not sure how we'll remember to do that 😄
> 
> Couldn't we just tag the applications v1 now? It wouldn't have any effect. Then when the new application flow comes out, make sure to set the version to v2.

Just want to make sure no new applications would be missed during that in-between period of unknown length, for the sake of our live data.

File: app/Models/JobApplication.php [hunk 1]
```diff
@@ -72,7 +74,8 @@ class JobApplication extends BaseModel
         'submission_signature' => 'string',
         'submission_date' => 'string',
         'experience_saved' => 'boolean',
-        'language_requirement_confirmed' => 'boolean'
+        'language_requirement_confirmed' => 'boolean',
+        'version_id'
```
Comment (yonikid15) [inline app/Models/JobApplication.php]: Forgot to add the type here. Otherwise looks good 👍

Comment (yonikid15) [reply app/Models/JobApplication.php]: Actually I can just add it in myself

**Comment by Khristophor** [comment]
After thinking about it, we can add a migration to set the existing posters to v1, as well as some logic to set each application to the "current" version (we can set this in a config file or in the application model directly) and then update that version variable once we roll out the new application flow.

**Pull request merged**